helpers/compile/plugins/fill-plugin/fillers/buffer-small.ts

```diff
@@ -0,0 +1,504 @@
+      }
+    }
+
+    return this
+  }
+
+  includes(value: string | number | Uint8Array, byteOffset = 0, encoding: Encoding = 'utf-8') {
+    return this.indexOf(value, byteOffset, encoding) !== -1
+  }
+
+  lastIndexOf(
+    value: string | number | Uint8Array,
+    byteOffsetOrEncoding: number | Encoding = 0,
+    encoding: Encoding = 'utf-8',
+  ) {
+    return this.indexOf(value, byteOffsetOrEncoding, encoding, true)
+  }
+
+  indexOf(
+    value: string | number | Uint8Array,
+    byteOffsetOrEncoding: number | Encoding = 0,
+    encoding: Encoding = 'utf-8',
+    lastIndexOf = false,
+  ) {
+    const method = lastIndexOf ? this.findLastIndex.bind(this) : this.findIndex.bind(this)
+    encoding = typeof byteOffsetOrEncoding === 'string' ? byteOffsetOrEncoding : encoding
+    const toSearch = Buffer.from(typeof value === 'number' ? [value] : value, encoding)
+    let byteOffset = typeof byteOffsetOrEncoding === 'string' ? 0 : byteOffsetOrEncoding
+    byteOffset = byteOffset < 0 ? this.length + byteOffset : byteOffset
+
+    if (toSearch.length === 0 && lastIndexOf === false) {
+      return byteOffset >= this.length ? this.length : byteOffset
+    }
+    if (toSearch.length === 0 && lastIndexOf === true) {
+      return (byteOffset >= this.length ? this.length : byteOffset) || this.length
+    }
+
+    return method((_, i) => {
+      const searchIf = lastIndexOf ? i <= (byteOffset || this.length) : i >= byteOffset
+      return searchIf && this[i] === toSearch[0] && toSearch.every((val, j) => this[i + j] === val)
+    })
+  }
+
+  toString(encoding: Encoding = 'utf8', start = 0, end = this.length) {
+    start = start < 0 ? 0 : start
+    encoding = encoding.toString().toLowerCase() as Encoding
+
+    if (end <= 0) return ''
+
+    if (encoding === 'utf8' || encoding === 'utf-8') {
+      return decoder.decode(this.slice(start, end))
+    }
+    if (encoding === 'base64' || encoding === 'base64url') {
+      const string = btoa(this.reduce((s, v) => s + c2s(v), ''))
+
+      if (encoding === 'base64url') {
+        return string.replace(/\+/g, '-').replace(/\//g, '_').replace(/=+$/, '')
+      }
+
+      return string
+    }
+    if (encoding === 'binary' || encoding === 'ascii' || encoding === 'latin1' || encoding === 'latin-1') {
+      return this.slice(start, end).reduce((s, v) => s + c2s(v & (encoding === 'ascii' ? 0x7f : 0xff)), '')
+    }
+    if (encoding === 'ucs2' || encoding === 'ucs-2' || encoding === 'utf16le' || encoding === 'utf-16le') {
+      let string = ''
+      const view = new DataView(this.slice(start, end).buffer)
+      for (let i = 0; i < view.byteLength; i += 2) {
+        string += c2s(view.getUint16(i, true))
+      }
+      return string
+    }
+    if (encoding === 'hex') {
+      return this.slice(start, end).reduce((s, v) => s + v.toString(16).padStart(2, '0'), '')
+    }
+
+    bufferPolyfillDoesNotImplement(`encoding "${encoding}"`)
```

Comment:
Would using `TextEncoder` here make sense? Or is it something `TextEncoder` does not support either?